### PR TITLE
Add HttpSender abstraction

### PR DESCRIPTION
**Pull request by jack-berg**
Part of #5351.

This is the first of several phases. Adds HttpSender interface, with OkHttpHttpSender implementation. All of this still lives in `opentelemetry-exporter-common` so this is just a refactor with no functional change at this point.

### CODE REVIEW

**Review by mateuszrzeszutek — APPROVED** [review]

File: exporters/common/src/main/java/io/opentelemetry/exporter/internal/http/HttpExporter.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+  public HttpExporter(
+      String exporterName,
+      String type,
+      HttpSender httpSender,
+      Supplier<MeterProvider> meterProviderSupplier,
+      boolean exportAsJson) {
+    this.type = type;
+    this.httpSender = httpSender;
+    this.exporterMetrics =
+        ExporterMetrics.createHttpProtobuf(exporterName, type, meterProviderSupplier);
```
[the remark this answers is not in the document]
Comment (jack-berg) [reply exporters/common/src/main/java/io/opentelemetry/exporter/internal/http/HttpExporter.java:51]: Good catch!

**Review by mateuszrzeszutek — APPROVED** [review]

File: exporters/common/src/main/java/io/opentelemetry/exporter/internal/http/HttpSender.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+/**
+ * An abstraction for sending HTTP requests and handling responses.
+ *
+ * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
+ * at any time.
+ *
+ * @see HttpExporter
+ * @see HttpExporterBuilder
+ */
+public interface HttpSender {
```
Comment (jkwatson) [inline exporters/common/src/main/java/io/opentelemetry/exporter/internal/http/HttpSender.java:22]: I've had to create this thing in multiple projects now. It's almost like the java ecosystem needs an http client equivalent of slf4j. 🤔

Comment (jkwatson) [reply exporters/common/src/main/java/io/opentelemetry/exporter/internal/http/HttpSender.java:22]: And be based on an interface, not an abstract class.  😿

**Review by jkwatson — APPROVED** [review]
Thanks!

**Pull request merged**